js/translator.js

```diff
@@ -40,9 +40,9 @@ function translate_story(nav) {
 
   check_lang();
 
-  url = location.href.replace(/[#\?\!]+.*/, "");
+  var myurl = location.href.replace(/[#\?\!]+.*/, "");
   permalink.style.display = '';
-  permalink.innerHTML = "<a href=\"" + url + "?" + idx + "\">Permalink to this story</a><span class=\"dot\"> • </span>";
+  permalink.innerHTML = "<a href=\"" + myurl + "#story-" + idx + "\">Permalink to this story</a><span class=\"dot\"> • </span>";
```

Comment:
Everything works perfectly, but I wouldn't mind changing the canonical url here from `myurl + "#story-" + idx` to just `myurl + "#" + idx`. This would help with e.g. localization etc and it would be easier I think to just have the number as a reference. I tested this with both patterns and the url change seems to work fine either way.

Also, since this is a permalink it might be a good idea to accept both `?` and `#` in the url (or at least redirect `?\d+` to `#\d+`), in case there are any existing links that would break as a result of the change. Off the top of my head I can't think of any such links, but I have a feeling there might be some on the African Storybook blog and other various places.